data/events/scripts/creature.lua

```diff
@@ -14,30 +14,22 @@ local function removeCombatProtection(cid)
 	end
 
 	player:setStorageValue(Global.Storage.CombatProtectionStorage, 2)
-	addEvent(function(cid)
-		local player = Player(cid)
-		if not player then
+	addEvent(function(playerUid)
```

Comment:
Change the "playerUid" variables that are inside the "function" body to another name. It could be something like "playerFuncUid" or something like that.